ABO/BankAccountParser.php

```diff
@@ -0,0 +1,145 @@
+        'Banka Creditas, a.s.' => '2250',
+    ];
+    private string $defaultBankCode;
+
+    /**
+     * @var array <string, string>
+     */
+    private array $bankAccountNumberToBankCodeMap;
+
+    /**
+     * @param string                $defaultBankCode
+     * @param array<string, string> $bankAccountNumberToBankCodeMap Allows mapping bank account to bank code (to
+     *                                                              construct correct full bank account for abo exports that does not contain bank
+     *                                                              code)
+     */
+    public function __construct(
+        string $defaultBankCode = '0000',
+        array $bankAccountNumberToBankCodeMap = []
+    ) {
+        $this->defaultBankCode = $defaultBankCode;
+        $this->bankAccountNumberToBankCodeMap = $bankAccountNumberToBankCodeMap;
+    }
+
+
+    /**
+     * Takes header line from ABO format and constructs bank account number. Line looks like:
+     * 0740000002100199001AA....00000170290722FIO and output looks like 2100199001/2010 or 115-214740207/0100 or
+     * 973789052/0000, etc.
+     *
+     * @see BankAccountParserTest
+     */
+    public function parse(string $line): string
+    {
+        $bankCode = $this->getBankCode(line: $line);
+
+        list($prefix, $number) = $this->getPrefixAndNumber(bankCode: $bankCode, line: $line);
+
+        $bankAccount = $this->formatBankAccountNumber(prefix: $prefix, number: $number);
+
+        if (array_key_exists($bankAccount, $this->bankAccountNumberToBankCodeMap)) {
+            $bankCode = $this->bankAccountNumberToBankCodeMap[$bankAccount];
+        }
+
+        return $bankAccount . '/' . $bankCode;
+    }
+
+    public function formatBankAccountNumber(string $prefix, string $number): string
+    {
+        // Remove leading zeros from prefix and number and build an array of non empty string values
+        $accountNumberParts = array_filter([
+            ltrim($prefix, '0'),
+            ltrim($number, '0')
+        ], fn(string $value) => $value !== '');
+
+        if ($accountNumberParts === []) {
+            $accountNumberParts = ['0000000000'];
+        }
+
+        // Implode will add '-' separator between prefix and number only if $prefix is not and empty string
+        // (built array will not contain empty prefix due the array_filter usage).
+        return implode('-', $accountNumberParts);
+    }
+
+    /**
+     * @param string $bankCode
+     * @param string $line
+     *
+     * @return array{0: string, 1: string}
+     */
+    protected function getPrefixAndNumber(string $bankCode, string $line): array
+    {
+        if (array_key_exists($bankCode, $this->bankCodesWithInternalFormat) === false) {
+            return [
+                substr($line, 3, 6),
+                substr($line, 9, 10)
+            ];
+        }
+
+        // Example: 7204021470000115
+        $internalAccountNumber = substr($line, 3, 16);
```

Comment:
I would move transformation of internal to normal format to separate utility function in ABOParser

Reply:
I tend to disagree. At this moment ABOParser is huge and the testability of the code is thank to it degreased. It is sensible to use the "service" class to handle bank account transformation so we can test it in detail in its own test case and ABOParser test will not care about implementation detail.